content/chainguard/chainguard-images/features/private-apk-repos/index.md

```diff
@@ -0,0 +1,319 @@
+
+```shell
+cat > Dockerfile <<EOF
+FROM cgr.dev/$ORGANIZATION/$IMAGE
+
+RUN apk add apko && apko install-keys
+RUN echo https://apk.cgr.dev/$ORGANIZATION >> /etc/apk/repositories
+RUN --mount=type=secret,id=cgr-token HTTP_AUTH="basic:apk.cgr.dev:user:\$(cat /run/secrets/cgr-token)" apk update
+RUN --mount=type=secret,id=cgr-token HTTP_AUTH="basic:apk.cgr.dev:user:\$(cat /run/secrets/cgr-token)" apk add wget -X https://apk.cgr.dev/$ORGANIZATION
+```
+
+Again, this Dockerfile includes the `-X` argument in the `apk add` command to ensure that it uses the private APK repository.
+
+Now you can build the image while passing along credentials obtained with `chainctl`. The following example builds an image named `my-custom-image`:
+
+```shell
+CGR_TOKEN=$(chainctl auth token --audience apk.cgr.dev) \
+   docker build --secret id=cgr-token,env=CGR_TOKEN . -t my-custom-image
+```
+
+Following that, you can verify that the image has access to the private repository and contains the apk package that was installed at build time:
+
+```shell
+docker run --rm my-custom-image apk policy wget
+```
+```Output
+wget policy:
+
+. . .
+
+  1.24.5-r4:
+	https://packages.wolfi.dev/os
+	https://apk.cgr.dev/$ORGANIZATION
+  1.24.5-r5:
+	https://packages.wolfi.dev/os
+	https://apk.cgr.dev/$ORGANIZATION
+  1.25.0-r0:
+	lib/apk/db/installed
+	https://packages.wolfi.dev/os
+	https://apk.cgr.dev/$ORGANIZATION
+```
+
+As this output shows, the `wget` apk package is installed in the container.
+
+
+## Using Private APK Repositories with Apko Builds
+
+You can also use your private APK repository with [apko](/open-source/build-tools/apko/overview/) builds. One of the advantages of this method is that you can build distroless images that include only the apk packages you need in the final image. 
+
+As with the previous examples, you'll need to provide the `HTTP_AUTH` environment variable containing your Chainguard token to the apko runtime building the image. 
+
+The following Dockerfile includes the private APK repository used in previous examples and installs a single package (`wget`) in the image:
+
+```shell
+cat > apko.yaml <<EOF
+contents:
+  repositories:
+  - https://apk.cgr.dev/$ORGANIZATION
+  keyring:
+  - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
+  packages:
+  - wget
+
+archs:
+- x86_64
+- aarch64
+EOF
+```
+
+The following `docker run` command will inject the `HTTP_AUTH` environment variable within an `apko` runtime and build the image:
+
+```shell
+docker run --rm \
+  -e "HTTP_AUTH=basic:apk.cgr.dev:user:$(chainctl auth token --audience apk.cgr.dev)" \
+  --workdir /work -v ${PWD}:/work \
+  cgr.dev/chainguard/apko build apko.yaml test-apk test-apk.tar \
+  --arch host
```

Comment:
Not sure why, but I'm getting this error

```
Error: unknown flag: --arch
```

Reply:
it works for me, but it also works without it. I'll just remove it.